### ISSUE TITLE
Documentation is missing important information

### ISSUE THREAD

**Issue by maximusk** [npm/npm#15925]
[This page](https://docs.npmjs.com/cli/install) on `install` command describes the following algorithm:

> load the existing node_modules tree from disk
> clone the tree
> fetch the package.json and assorted metadata and add it to the clone
> walk the clone and add any missing dependencies
>   dependencies will be added as close to the top as is possible
>   without breaking any other modules
> compare the original tree with the cloned tree and make a list of
> actions to take to convert one to the other
> execute all of the actions, deepest first
>   kinds of actions are install, update, remove and move

> For this package{dep} structure: A{B,C}, B{C}, C{D}, this algorithm produces:

However, the documentation doesn't mention recursive nature of the algorithm. Is the same algorithm applied to `B` and consequently to `C` in the dependency chain mentioned above?

**Comment by legodude17** [comment]
@maximusk not really, because `npm` tries to flatten all the deps as much as possible, so adding `B ` and `C`'s dependencies are included in walking the tree.

**Comment by legodude17** [comment]
@maximusk not really, because `npm` tries to flatten all the deps as much as possible, so adding `B ` and `C`'s dependencies are included in walking the tree.

**Comment by maximusk** [comment]
@legodude17, hey, thanks for your feedback. But unfortunately I can't see how dependencies can be flattened before loading a see and finding out that D is required, Can you please clarify?

**Comment by legodude17** [comment]
Ah, I think I misunderstood the question. When `npm` is walking the tree, it is traversing all of the dependencies of everything to build a full dep tree.
